src/test/java/org/fogbowcloud/manager/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -0,0 +1,359 @@
+        
+        when(cloudPluginsHolder.getComputePlugin()).thenReturn(computePlugin);
+        when(cloudPluginsHolder.getAttachmentPlugin()).thenReturn(attachmentPlugin);
+        when(cloudPluginsHolder.getNetworkPlugin()).thenReturn(networkPlugin);
+        when(cloudPluginsHolder.getVolumePlugin()).thenReturn(volumePlugin);
+        when(cloudPluginsHolder.getImagePlugin()).thenReturn(imagePlugin);
+        when(cloudPluginsHolder.getComputeQuotaPlugin()).thenReturn(computeQuotaPlugin);
+        when(aaController.getLocalToken(any(FederationUser.class))).thenReturn(new Token());
+        
+        this.localCloudConnector = new LocalCloudConnector(aaController, cloudPluginsHolder);
+        
+    }
+	
+	@Test
+	public void testRequestComputeInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(ComputeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    when(computePlugin.requestInstance(any(ComputeOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
+	}
+	
+	@Test
+	public void testRequestAttachmentInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(AttachmentOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.ATTACHMENT);
+	    when(attachmentPlugin.requestInstance(any(AttachmentOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
+	}
+	
+	@Test
+	public void testRequestVolumeInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(VolumeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.VOLUME);
+	    when(volumePlugin.requestInstance(any(VolumeOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
+	}
+	
+	@Test
+	public void testRequestNetworkInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(NetworkOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.NETWORK);
+	    when(networkPlugin.requestInstance(any(NetworkOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
+	}
+	
+	// If plugin returns a null instance id, the method requestInstance() must throw an exception
+	@Test(expected = UnexpectedException.class)
+	public void testExceptionNullInstanceId() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(ComputeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    when(computePlugin.requestInstance(any(ComputeOrder.class), any(Token.class))).thenReturn(null);
+		this.localCloudConnector.requestInstance(order);
+	}
+	
+	
```

Comment:
follow camel case ...

Reply:
also, please, test this nullity-exception with the order plugins, not only with the compute one

Reply:
Done